addons/minedetector/functions/fnc_playDetectorSound.sqf

```diff
@@ -27,6 +26,9 @@ if (!alive _unit) exitWith {
     ACE_LOGERROR_1("unit is not alive [%1]",_unit);
 };
 
-private _posASL = AGLtoASL (_unit modelToWorld (_unit selectionPosition "granat"));
-
-playSound3D [_sound, objNull, false, _posASL, 5, _pitch, 15];
+if (_unit getVariable [QGVAR(isUsingHeadphones), false]) then {
+    _unit say2D _soundClass;
```

Comment:
That may do the same. At least the queuing. It's why I used a dummy object for the sound.